tests/RobotFramework/tests/tedge/certificate_signing_request.robot

```diff
@@ -85,6 +85,26 @@ Generate CSR using the device-id from an existing certificate and private key of
     Should Be Equal    ${hash_before_cert}    ${hash_after_cert}
     Should Be Equal    ${hash_before_private_key}    ${hash_after_private_key}
 
+Generate CSR using the CSR path from tedge config
+    [Setup]    Setup Without Certificate
+    File Should Not Exist    /etc/tedge/device-certs/tedge-certificate.pem
+    File Should Not Exist    /etc/tedge/device-certs/tedge-private-key.pem
+
+    Execute Command
+    ...    tedge config set c8y.device.csr_path /etc/tedge/device-certs/example_dev_test0001-test1.csr
+
+    Execute Command    sudo tedge cert create-csr
```

Comment:
For the failing test, the error message makes sense. Since there is no device certificate nor explicitly device ID is provided by `--device-id` option, there is no way for the command to know the CN.
So, either provide a device ID as the error message says, or create a certificate beforehand.

> sudo tedge cert create-csr returned an unexpected exit code stdout: stderr: Error: missing configuration parameter Caused by: No device ID is provided. Use --device-id &lt;name&gt; option to specify the device ID.

```suggestion
    Execute Command    sudo tedge cert create-csr --device-id test-user
```

Reply:
Is this a new test though? If not, then the `--device-id` is not a mandatory flag, as it can pick up the value from the existing certificate.